recipe_scrapers/marleyspoon.py

```diff
@@ -0,0 +1,76 @@
+import re
+
+from ._abstract import AbstractScraper
+from ._utils import get_minutes, normalize_string
+
+
+class MarleySpoon(AbstractScraper):
+    @classmethod
+    def host(cls, domain="de"):
+        return f"marleyspoon.{domain}"
+
+    def canonical_url(self):
+        return self.soup.find("meta", {"property": "og:url"})["content"]
+
+    def author(self):
+        return normalize_string(self.soup.find(class_="author-info").a.get_text())
+
+    def title(self):
+        return normalize_string(
+            self.soup.find("h3", {"class": "recipe-title"}).get_text()
+        )
+
+    def _specs(self):
+        return self.soup.find("ul", {"class": "recipe-specs"}).find_all("li")
```

Comment:
I'd probably suggest a refactor of this to extract each `recipe-spec` component into a named variable, and then to return those as an ordered tuple (or perhaps dictionary / other datastructure).

Something like:

```suggestion
    def _specs(self):
    def _specs(self):
        recipe_specs = self.soup.find("ul", {"class": "recipe-specs"})
        time_label = recipe_specs.find("label", string="Dauer")
        category_label = recipe_specs.find("label", string="Schwierigkeitsgrad")
        nutrient_label = recipe_specs.find("label", string="Nährwertangaben pro Portion")
        time_info = time_label.find_next_sibling("p")
        category_info = category_label.find_next_sibling("p")
        nutrient_info = nutrient_label.find_next_sibling("p")
        return time_info, category_info, nutrient_info
```

(that's a rough draft; clarity/conciseness improvements are likely possible)

Reply:
good idea. Would you prefer a solution involving a vocabulary of the terms in different languages here as well or would just counting the `<li>` be sufficient?

Reply:
I'd prefer to use per-language vocabularies, yep - counting can be a bit fragile, particularly over time -- one item might be removed, and then an unrelated one added, for example.

That does imply that in theory, we're adding code that might not handle some languages for recipes on the site (either for existing recipe pages, or ones added in future).  Perhaps in practice most/all of them currently match the corresponding host domain-name.. even so I think it may make sense to raise an exception (`UnsupportedLocaleException`?) if no vocab is found for the page language.